File: C# (ASP.NET & WPF)/C# - INSY7315 - EliteRentals - (Web App, Mobile App & API)/Elite-Rentals-Rest-API/EliteRentalsAPI/Models/Property.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EliteRentalsAPI.Models
{
    public class Property
    {
        [Key]
        public int PropertyId { get; set; }

        [ForeignKey("Manager")]
        public int ManagerId { get; set; }
        public User? Manager { get; set; }   // Navigation

        public string Title { get; set; } = "";
        public string Description { get; set; } = "";
        public string Address { get; set; } = "";
        public string City { get; set; } = "";
        public string Province { get; set; } = "";
        public string Country { get; set; } = "";
        public decimal RentAmount { get; set; }
        public int NumOfBedrooms { get; set; }
        public int NumOfBathrooms { get; set; }
        public string ParkingType { get; set; } = "";
        public int NumOfParkingSpots { get; set; }
        public bool PetFriendly { get; set; }
        public string Status { get; set; } = "Available";
        public DateTime ListingDate { get; set; } = DateTime.UtcNow;

        /*        public byte[]? ImageData { get; set; }
                public string? ImageType { get; set; }*/

        public ICollection<PropertyImage>? Images { get; set; }

        //Leases navigation
        public ICollection<Lease>? Leases { get; set; }
    }
}

File: Kotlin/Kotlin - PROG7314 - EliteRentals (Mobile App & API Development)/Elite-Rentals-Mobile-App/app/src/main/java/com/rentals/eliterentals/Property.kt
package com.rentals.eliterentals

import java.math.BigDecimal

data class Property(
    val title: String,
    val description: String,
    val address: String,
    val city: String,
    val province: String,
    val country: String,
    val rentAmount: BigDecimal,
    val numOfBedrooms: Int,
    val numOfBathrooms: Int,
    val parkingType: String,
    val numOfParkingSpots: Int,
    val petFriendly: Boolean,
    val status: String
)
